### PR TITLE
fix bug: COM_EXECUTE in MySQL protocol will response by binary row not text row

### PR DESCRIPTION
**Pull request by ouyuanning**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5017 #4994 

## What this PR does / why we need it:
fix bug: COM_EXECUTE in MySQL protocol will response by binary row not text row

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
Missing tests.

File: pkg/frontend/mysql_protocol.go [hunk 1]
```diff
@@ -1631,6 +1631,127 @@ func (mp *MysqlProtocolImpl) sendColumns(mrs *MysqlResultSet, cmd int, warnings,
 	return nil
 }
 
+// the server convert every row of the result set into the format that mysql protocol needs
+func (mp *MysqlProtocolImpl) makeResultSetBinaryRow(buffer []byte, mrs *MysqlResultSet, rowIdx uint64) ([]byte, error) {
+	buffer = append(buffer, defines.OKHeader) // append OkHeader
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:1636]: remove the buffer. 
use mp.appendStringLenEnc which maintain the internal buffer for you.

Comment (daviszhen) [reply pkg/frontend/mysql_protocol.go:1636]: in makeResultSetTextRow , it use a lot of mp.appendXXX which maintain the network buffer.

**Review by daviszhen — CHANGES_REQUESTED** [review]

**Comment by ouyuanning** [comment]
> Missing tests.

ut was added.
bvt can not test this code.  you must have a jdbc client with prepareStatement to run SQL.

this bug was found by @aressu1985 when he develop TPCC test code in java.   
@aressu1985 said he will check by his TPCC test code for me.  So a good guy.

File: pkg/frontend/mysql_protocol.go [hunk 2]
```diff
@@ -1631,6 +1649,129 @@ func (mp *MysqlProtocolImpl) sendColumns(mrs *MysqlResultSet, cmd int, warnings,
 	return nil
 }
 
+// the server convert every row of the result set into the format that mysql protocol needs
+func (mp *MysqlProtocolImpl) makeResultSetBinaryRow(data []byte, mrs *MysqlResultSet, rowIdx uint64) ([]byte, error) {
+	data = mp.append(data, defines.OKHeader) // append OkHeader
+
+	// get null buffer
+	var buffer []byte
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:1657]: reuse the buffer just like lenencbuffer.

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/mysql_protocol.go [hunk 3]
```diff
@@ -1631,6 +1649,128 @@ func (mp *MysqlProtocolImpl) sendColumns(mrs *MysqlResultSet, cmd int, warnings,
 	return nil
 }
 
+// the server convert every row of the result set into the format that mysql protocol needs
+func (mp *MysqlProtocolImpl) makeResultSetBinaryRow(data []byte, mrs *MysqlResultSet, rowIdx uint64) ([]byte, error) {
+	data = mp.append(data, defines.OKHeader) // append OkHeader
+
+	// get null buffer
+	columnsLength := mrs.GetColumnCount()
+	buffer := mp.lenEncBuffer[:columnsLength]
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:1658]: the lenEncBuffer only has 10 elements.

**Review by daviszhen — APPROVED** [review]

**Pull request merged**